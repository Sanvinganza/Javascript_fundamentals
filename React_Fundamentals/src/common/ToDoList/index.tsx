import * as React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import TodoInput from './redux/components/TodoInput';
import TodoList from './redux/components/TodoList';
import rootReducer from './redux/reducers';

export default function ToDoList () {
  const store = createStore(rootReducer);

  return (
    <Provider store={store}>
      <TodoInput />
      <TodoList />
    </Provider>
  );
}
