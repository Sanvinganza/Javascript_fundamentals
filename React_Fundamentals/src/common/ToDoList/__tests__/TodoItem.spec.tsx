import { render, screen, waitFor } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import ToDoList from ".."

describe('TodoItem component', () => {
  test('render is correct', () => {
    expect(render(<ToDoList />)).toMatchSnapshot();
  });
  test('function newTodo should worked corrent', async () => {
    const newTodoButton = screen.getByText('newTodo');
    expect(newTodoButton).toBe('button');
    await waitFor(() => userEvent.click(newTodoButton));

    screen.debug
  });
})