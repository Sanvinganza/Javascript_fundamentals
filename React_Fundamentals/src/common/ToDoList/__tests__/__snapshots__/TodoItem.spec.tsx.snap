// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem component render is correct 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo"
      >
        <form>
          <input
            name="todo"
            placeholder="add a todo item"
            type="text"
            value=""
          />
        </form>
        <div
          class="todo-list"
        >
          List is empty...
          <div
            class="footer"
          >
            <div
              class="items-left"
            >
              0
               items left
            </div>
            <button
              class="getTodo"
            >
              newTodo
            </button>
            <select
              name="select"
            >
              <option
                value="All"
              >
                All
              </option>
              <option
                value="Completed"
              >
                Completed
              </option>
              <option
                value="Not_Completed"
              >
                Not Completed
              </option>
            </select>
            <button
              type="button"
            >
              clear list
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo"
    >
      <form>
        <input
          name="todo"
          placeholder="add a todo item"
          type="text"
          value=""
        />
      </form>
      <div
        class="todo-list"
      >
        List is empty...
        <div
          class="footer"
        >
          <div
            class="items-left"
          >
            0
             items left
          </div>
          <button
            class="getTodo"
          >
            newTodo
          </button>
          <select
            name="select"
          >
            <option
              value="All"
            >
              All
            </option>
            <option
              value="Completed"
            >
              Completed
            </option>
            <option
              value="Not_Completed"
            >
              Not Completed
            </option>
          </select>
          <button
            type="button"
          >
            clear list
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
